modules/integration/tests-integration/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/classMediator/PropertyPersistenceAddingTestCase.java

```diff
@@ -88,6 +88,8 @@ public void testMediationPropertyPersistenceAdding() throws Exception {
         serverConfigurationManager.copyToComponentLib
                 (new File(getClass().getResource(JAR_LOCATION + File.separator + CLASS_JAR_FIVE_PROPERTIES).toURI()));
         loadSampleESBConfiguration(0);
+        /* waiting for the new config file to be written to the disk */
+        Thread.sleep(10000);
```

Comment:
Is this a correct way to solve the actual issue? Introducing a sleep is not the optimal way. Can you explain how this can solve the actual issue?